codegen/masm/intrinsics/mem.masm

```diff
@@ -336,31 +342,29 @@ export.realign_dw # [chunk_hi, chunk_mid, chunk_lo, offset]
     swap.1 # [x_hi, x_lo]
```

Comment:
@bitwalker I think the issue is that the hi and lo parts of the value are in reverse order of what is expected by the test assertion. I see it on the stack after the last `swap.1` instruction in the `realign_dw` function. I have a gut feeling that there is either one extra hi-lo parts swap along the `load/store` way or one is missing.

Reply:
The argument values for `realign_dw` has mid and lo parts swapped. I believe this is due to the `store_dw` target memory layout (`# target memory layout: [hi, ..], [..lo, mid]`) in the last `else` block differs from the expected memory layout in `load_dw` (`# memory layout: [hi, ..], [<unused>, <unused>, mid, lo]`). The `mid` and `lo` are swapped.

Reply:
Fixed in https://github.com/0xPolygonMiden/compiler/pull/284/commits/c010baeb6433c406055917ba45bdc8236e2444ab

Reply:
Nice catch! Looks like that fixes it, I'll follow up once I've validated `get_inputs` and `blake3` tests!

Reply:
I've verified (and fixed up/re-enabled the `get_inputs` test), looking into blake3 now